### PR TITLE
Add AnimatableTabBarController to support custom transition animation

### PR DESCRIPTION
**Pull request by phimage**
Same as `AnimatableNavigationController` with reverse animation when we can using child controller index

Added to the demo too, near the other tab bar controller where the animation is coded, not configured using storyboard

### CODE REVIEW

File: IBAnimatable/Navigator.swift [hunk 1]
```diff
@@ -74,3 +75,43 @@ extension Navigator: UINavigationControllerDelegate {
...
+
+    guard let viewControllers = tabBarController.viewControllers,
+      let fromVCIndex = viewControllers.index(of: fromVC),
+      let toVCIndex = viewControllers.index(of: toVC) else {
+        return nil
+    }
+    if toVCIndex < fromVCIndex {
+      animator?.transitionDuration = transitionDuration
+      return animator
+    } else {
```
Comment (tbaranes) [inline IBAnimatable/Navigator.swift]: The else is not mandatory since the if returns

File: IBAnimatable/Navigator.swift [hunk 2]
```diff
@@ -74,3 +75,43 @@ extension Navigator: UINavigationControllerDelegate {
...
+    }
+  }
+
+  // MARK: - interaction controller
+  public func tabBarController(_ tabBarController: UITabBarController,
+                               interactionControllerFor animationController: UIViewControllerAnimatedTransitioning)
+    -> UIViewControllerInteractiveTransitioning? {
+      if let interactiveAnimator = interactiveAnimator, interactiveAnimator.interacting {
+        return interactiveAnimator
+      } else {
```
Comment (tbaranes) [inline IBAnimatable/Navigator.swift]: Same here, you can drop the else (if that's a copy from elsewhere in the code, you can remove it there as well)

**Review by tbaranes — CHANGES_REQUESTED** [review]
Looks good, I just let you a small a few code style comments 👍 

Can you also update the CHANGELOG?

**Comment by tbaranes** [comment]
_The following comments is not directly related to this PR_

I'm wondering, should we move the both tabBar examples in the `Transition` part (by diving in 3 subnavigations) directly?

File: IBAnimatable/Navigator.swift (hunk 2 above)
Comment (phimage) [reply IBAnimatable/Navigator.swift]: Yes it's a copy paste like always, I am a copy past coder 😉 

here I think I could do
```swift
interactiveAnimator?.interacting
```
edit/ my mistake I don`t watch carefully the code. I will just remove the else

File: IBAnimatable/Navigator.swift (hunk 1 above)
Comment (phimage) [reply IBAnimatable/Navigator.swift]: I can simplify the code here, and I will remove unnecessary else if there is some return

File: IBAnimatable/Navigator.swift (hunk 2 above)
Comment (tbaranes) [reply IBAnimatable/Navigator.swift]: Even better! Best way to conform to designable protocols: copy 😬

**Review by tbaranes — APPROVED** [review]

**Review by JakeLin — APPROVED** [review]
Great PR. Animatable everything 💪

**Comment by JakeLin** [comment]
@phimage Great PR 👍👍👍, can you merge it when you feel it is ready?

**Pull request merged**